pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -237,6 +245,14 @@ func (ku *Upgrader) upgradeAgentPools(ctx context.Context) error {
 		transformer := &transform.Transformer{
 			Translator: ku.Translator,
 		}
+
+		if ku.ClusterTopology.DataModel.Properties.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision() {
+			err = transformer.RemoveJumpboxResourcesFromTemplate(ku.logger, templateMap)
+			if err != nil {
+				return ku.Translator.Errorf("error removing jumpbox resources from template: %s", err.Error())
```

Comment:
This code will never be reached with current implementation since it only warns about the expected errors.

Reply:
Maybe that's ok, I'm basically just confirming that we want to just warn when the jumpbox JSON is invalid, rather than emit an error.

Reply:
This is a good catch. I've modified the `RemoveJumpboxResourcesFromTemplate` method so that it returns an error if it isn't able to interpret any resource item in the interface-coerced resources array as a `map[string]interface{}`, because that's probably bad.

The resource not having a name property seems more appropriate to keep as a warning, perhaps there's an Azure resource spec somewhere without a name property?